frontend/src/store/FamLoginUserState.ts

```diff
@@ -215,66 +228,93 @@ const getCachedAppRolesForDelegatedAdmin = (
+        grant.roles?.forEach((role: FamRoleDto) => {
+            const roleDescription = 'Delegated Admin, ' + role.name;
+            if (!role.forest_clients) {
+                permissions.push({
                     application: grant.application.description,
                     env: grant.application.env,
+                    role: roleDescription,
                 });
-            });
-        }
-
-        if (item.auth_key === AdminRoleAuthGroup.DelegatedAdmin) {
-            item.grants.forEach((grant) => {
-                grant.roles?.forEach((role) => {
-                    if(!role.forest_clients) {
-                        myPermissions.push({
-                            application: grant.application.description,
-                            env: grant.application.env,
-                            clientId: null,
-                            role: 'Delegated Admin, ' + role.name,
-                        });
-                    } else {
-                        role.forest_clients?.forEach((clientId) => {
-                            myPermissions.push({
-                                application: grant.application.description,
-                                env: grant.application.env,
-                                clientId: clientId,
-                                role: 'Delegated Admin, ' + role.name,
-                            });
-                        });
-                    }
+            } else {
+                role.forest_clients.forEach((clientId: any) => {
+                    permissions.push({
+                        application: grant.application.description,
+                        env: grant.application.env,
+                        clientId: clientId,
+                        role: roleDescription,
+                    });
                 });
-            });
+            }
+        });
+    });
+
+    return permissions;
+};
+
+const getMyAdminPermission = (): IMyPermission[] => {
+    const myPermissions: IMyPermission[] = [];
+
+    getUserAccess()?.forEach((access: FamAuthGrantDto) => {
+        switch (access.auth_key) {
+            case AdminRoleAuthGroup.FamAdmin:
+                myPermissions.push(...getMyFamAdminPermission(access));
+                break;
+            case AdminRoleAuthGroup.AppAdmin:
+                myPermissions.push(...getMyAppAdminPermission(access));
+                break;
+            case AdminRoleAuthGroup.DelegatedAdmin:
+                myPermissions.push(...getMyDelegatedAdminPermission(access));
+                break;
+            default:
+                break;
         }
     });
 
     return myPermissions.map((permission: IMyPermission) => {
-
-        permission.application = permission.application.replace(/\([^()]*\)/g, '')
-        return permission
+        permission.application = permission.application!.replace(
+            /\([^()]*\)/g,
+            ''
+        );
+        return permission;
     });
 };
+
```

Comment:
Hmm... all above changed methods seems to me quite specific for only one component as it takes current cached user accesses and re-mapping them into specific structure needs on display.
Maybe it was discussed before and felt this is the file to add these methods here, otherwise I am not really sure if this state file is really the place for it (as initially the first time looking at the method, "getMyAdminPermission", I was confused for its purpose and object structure.)
I will skip for this. Just a comment.

Reply:
I agree Ian, we can move it out later

Reply:
sounds good!